crewai_tools/adapters/mcp_adapter.py

```diff
@@ -0,0 +1,97 @@
+"""
+MCPServer for CrewAI.
+
+
+"""
+
+from typing import Any
+
+from crewai.tools import BaseTool
+
+
+class MCPServerAdapter:
+    """Manages the lifecycle of an MCP server and make its tools available to CrewAI.
+
+    Note: tools can only be accessed after the server has been started with the
+        `start()` method.
+
+    Attributes:
+        tools: The CrewAI tools available from the MCP server.
+
+    Usage:
+        # context manager + stdio
+        with MCPServerAdapter(...) as tools:
+            # tools is now available
+
+        # context manager + sse
+        with MCPServerAdapter({"url": "http://localhost:8000/sse"}) as tools:
+            # tools is now available
+
+        # manually start / stop mcp server
+        mcp_server = MCPServerAdapter(...)
+        tools = mcp_server.tools
+        mcp_server.stop()
```

Comment:
I'm a bit concerned about this usage.. what about recommending to use try/finally when not using a manage context? So we ensure "stop" even when an error throw

Reply:
yes fully agree